app/models/two_factors_token.rb

```diff
@@ -0,0 +1,16 @@
+class TwoFactorsToken < ApplicationRecord
+  belongs_to :user
+
+  scope :active, -> do
+    where("created_at >= ?", self.active_duration.ago)
```

Comment:
We use Arel in the warehouse to maintain some small amount of database agnostic-ness.  This can be written `where(arel_table[:created_at].gteq(active_duration.ago))`.  Though, after the refactor, it will just be `where(are_table[:expires_at].gt(Time.current))`